starlite/middleware/csrf.py

```diff
@@ -0,0 +1,103 @@
+import hashlib
+import hmac
+import secrets
+from http.cookies import SimpleCookie
+from typing import Optional
+
+from starlette.datastructures import MutableHeaders
+from starlette.responses import PlainTextResponse
+from starlette.types import ASGIApp, Message, Receive, Scope, Send  # noqa: TC002
+
+from starlite.config import CSRFConfig  # noqa: TC001
+from starlite.connection import Request
+from starlite.types import MiddlewareProtocol
+
+CSRF_SECRET_BYTES = 32
+CSRF_SECRET_LENGTH = CSRF_SECRET_BYTES * 2
+
+
+class BadSignature(Exception):
+    pass
+
+
+class CSRFMiddleware(MiddlewareProtocol):
+    def __init__(
+        self,
+        app: ASGIApp,
+        config: CSRFConfig,
+    ):
+        super().__init__(app)
+        self.app = app
+        self.config = config
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        if scope["type"] != "http":
+            await self.app(scope, receive, send)
+            return
+
+        request: Request = Request(scope=scope)
```

Comment:
```suggestion
        request = Request(scope=scope)
```

Reply:
When I remove the type hint `mypy` complains 🤦‍♂️